docs/using_eessi/building_on_eessi.md

```diff
@@ -0,0 +1,69 @@
+# Building software on top of EESSI
+
+## Building software on top of EESSI with EasyBuild
+Building on top of EESSI with EasyBuild is relatively straightforward. One crucial feature is that EasyBuild supports building against operating system libraries that are _not_ in a standard prefix (such as `/usr/lib`). This is required when building against EESSI, since all of the software in EESSI is build against the [compatibility layer](../compatibility_layer.md).
+
+### Starting the EESSI software environment
+Start your environment as described [here](../using_eessi/setting_up_environment.md)
+
+### Configure EasyBuild
+To configure EasyBuild, first, check out the Github repository. We advice you check out the branch corresponding to the version of EESSI you would like to use.
+
+If you are unsure which version you are using, you can run
+```
+echo ${EESSI_PILOT_VERSION}
+```
+to check it.
+
+To build on top of e.g. version `2023.06` of the EESSI software stack, we check it out, and go into that directory:
+
+```
+git clone https://github.com/EESSI/software-layer/ --branch 2023.06
+cd software-layer
+```
+Then, you have to pick a working directory (that you have write access to) where EasyBuild can do the build, and an install directory, where EasyBuild can install it. In this example, we use `/tmp/easybuild` as our working directory, and `$HOME/.local/easybuild` as our installpath:
+```
+export WORKDIR=/tmp/easybuild
+export EASYBUILD_INSTALLPATH="${HOME}/.local/easybuild"
+source configure_easybuild
+```
+Next, you load the EasyBuild module that you want to use, e.g. 
+```
+module load EasyBuild/4.8.1
+```
+Finally, you can check the current configuration for EasyBuild using
+```
+eb --show-config
+```
+
+!!! Note
+    We use EasyBuild's default behaviour in optimizing for the host architecture. Since the EESSI initialization script also loads the EESSI stack that is optimized for your host architecture, this matches nicely. However, if you work on a cluster with heterogeneous node types, you have to realize you can only use these builds on the same architecture as where you build them. You can use different `EASYBUILD_INSTALLPATH`s if you want to build for different host architectures.
```

Comment:
Hmm, if I had a `zen4` system what would happen? Would I build for `zen4` or `zen3` (or what the best fit is)?

Reply:
I'm not really sure what you are asking here. If you are building on a `zen4` host, you optimize for that, since EasyBuild's default is to optimize for the build host. 

I'll make the note more elaborate with a concrete example though.